chain/traversal.go

```diff
@@ -59,6 +58,12 @@ func (it *TipsetIterator) Complete() bool {
 
 // Next advances the iterator to the next value.
 func (it *TipsetIterator) Next() error {
-	it.value, it.err = GetParentTipSet(it.ctx, it.store, it.value)
-	return it.err
+	var err error
+	select {
+	case <-it.ctx.Done():
+		return it.ctx.Err()
+	default:
+		it.value, err = GetParentTipSet(it.ctx, it.store, it.value)
+	}
+	return err
```

Comment:
@rosalinekarr please propose and land this independent change in a separate PR and rebase onto it. The iterator has tests and they should be extended to exercise this branch.

Landing independent changes ahead of a larger PR is a good pattern for keeping reviews manageable. I call it a "prefactor".

Reply:
Done: https://github.com/filecoin-project/go-filecoin/pull/2544